packages/matter.js/src/device/Endpoint.ts

```diff
@@ -53,9 +62,22 @@ export class Endpoint {
         this.addClusterServer(this.descriptorCluster);
         this.setDeviceTypes(deviceTypes);
 
-        if (endpointId !== undefined) {
-            this.id = endpointId;
+        if (options.endpointId !== undefined) {
+            this.id = options.endpointId;
         }
+        if (options.uniqueId !== undefined) {
+            this.uniqueId = options.uniqueId;
+        }
+    }
+
+    setStructureChangedCallback(callback: () => void) {
+        this.structureChangedCallback = callback;
+        this.childEndpoints.forEach(endpoint => endpoint.setStructureChangedCallback(callback));
```

Comment:
Could these events bubble up from child endpoints to their parent automatically?  Might simplify state management.  Not sure there's a huge advantage but that's you wouldn't need the forEach here

Reply:
For now we need to update too many data (partsLists and full list of endpoints and attributes), so thats why I decided to just inform the Commissioning server whoch then does it "down the structure" again. So bubbling through the structure would bring no benefit here.